.github/workflows/wheels.yml

```diff
@@ -0,0 +1,80 @@
+name: Wheel build
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    paths:
+      - pyproject.toml
+      - setup.py
+      - .github/workflows/wheels.yml
+
+# When this workflow is queued, automatically cancel any previous running
+# or pending jobs from the same branch
+concurrency:
+  group: wheels-${{ github.head_ref }}
+  cancel-in-progress: true
+
+jobs:
+  wheels:
+    name: Build (and upload)
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3.5.0
+        with:
+          fetch-depth: 0
+      - name: Install dependencies
+        run: |
+          pip install build
+
+          which python
+          pip list
+      - name: Get current and new versions
+        run: |
+          # suffix for nightly wheel versions
+          suffix=a`date +%y%m%d`
+
+          # get current version
+          version=$(git describe --tags --abbrev=0)
+
+          # construct new version
+          major=$(echo "$version" | cut -d. -f1)
+          minor=$(echo "$version" | cut -d. -f2)
+          patch=$(echo "$version" | cut -d. -f3)
+          patch=$((patch + 1))
+
+          new_version="$major.$minor.$patch"
+
+          # export to GHA env
+          echo old_version=${version} >> $GITHUB_ENV
+          echo new_version=${new_version} >> $GITHUB_ENV
+      - name: Update Distributed pinning
+        uses: jacobtomlinson/gha-find-replace@v3
+        with:
+          include: 'pyproject.toml'
+          find: "== ${{ env.old_version }}"
+          replace: ">= ${{ env.old_version }},<${{ env.new_version }}b"
+          regex: false
+      - name: Build wheel
+        run: |
+          # commit changes
+          git config --local user.email "action@github.com"
+          git config --local user.name "GitHub Action"
+          git commit -am "Update distributed pinning"
+
+          # create new nightly tag
+          tag=${new_version}a`date +%y%m%d`
```

Comment:
Can/should we include a commit hash / build number here so that nightlies published on the same day don't collide? Followed the general version scheme of [dask-cuda](https://pypi.org/project/dask-cuda/) pre-releases, but this is inconsistent with our conda nightlies.

Reply:
It looks like these tags aren't persisted to the repo, they're only created locally and discarded at the end of the run. Is this because dask uses something like setuptools_scm for version management in releases?

Reply:
This is especially relevant so that we know what version pinning strategy to leverage in our rapids packages

Reply:
@jakirkham might know more about the version management, but my understanding as to why the tags aren't being persisted is because we aren't pushing them up to the repo with something like

```
git push https://github.com/dask/dask main --tags
```

Do we want to keep the nightly tags persisted? My main reason for avoiding this was just circumventing permissions I assume the runner needs to do this, but I'm also assuming we'd get some pushback if we wanted to push all the `2023.3.3a*` tags to the repo for sake of diluting the release history

Reply:
Sorry, I wasn't clear in what I was asking. _I do not want to persist the tags_. I am asking why we create them in the first place. Yes, the tags aren't persisted because they aren't pushed. My question was to understand why we create ephemeral tags in this way, and my theory was that we need them for some package like `setuptools_scm` that is used to manage package versions by looking at Git tags.